src/Standards/Generic/Sniffs/ControlStructures/ControlStructuresBracketsNewLineSniff.php

```diff
@@ -0,0 +1,228 @@
+            $prevContent = $tokens[($openBrace - 1)]['content'];
+
+            if ($prevContent === $phpcsFile->eolChar) {
+                $spaces = 0;
+            } else {
+                $blankSpace = substr($prevContent, strpos($prevContent, $phpcsFile->eolChar));
+                $spaces     = 0;
+
+                // A tab is only counted with strlen as 1 character but we want to count,
+                // the number of spaces so add 4 characters for a tab otherwise the strlen.
+                for ($i = 0; $length = strlen($blankSpace), $i < $length; $i++) {
+                    if ($blankSpace[$i] === "\t") {
+                        $spaces += $this->indent;
+                    } else {
+                        $spaces += strlen($blankSpace[$i]);
+                    }
+                }
+            }//end if
+
+            $nested = 0;
+
+            // Take into account any nested parenthesis that don't contribute to the level (often required for
+            // closures and anonymous classes).
+            if (array_key_exists('nested_parenthesis', $tokens[$stackPtr]) === true) {
+                $nested = count($tokens[$stackPtr]['nested_parenthesis']);
+            }//end if
+
+            $expected = (($tokens[$stackPtr]['level'] + $nested) * $this->indent);
+
+            // We need to divide by 4 here since there is a space vs tab intent in the check vs token.
+            $expected /= $this->indent;
+            $spaces   /= $this->indent;
+
+            if ($spaces !== $expected) {
+                $error = 'Expected %s tabs before opening brace; %s found';
+                $data  = [
+                    $expected,
+                    $spaces,
+                ];
+                $fix   = $phpcsFile->addFixableError($error, $openBrace, 'SpaceBeforeBrace', $data);
+
+                if ($fix === true) {
+                    $indent = str_repeat("\t", $expected);
+
+                    if ($spaces === 0) {
+                        $phpcsFile->fixer->addContentBefore($openBrace, $indent);
+                    } else {
+                        $phpcsFile->fixer->replaceToken(($openBrace - 1), $indent);
+                    }
+                }
+            }//end if
+        }//end if
+
+        // A single newline after opening brace (i.e. brace in on a line by itself), remove extra newlines.
+        if (isset($tokens[$stackPtr]['scope_opener']) === true) {
+            $opener = $tokens[$stackPtr]['scope_opener'];
+
+            for ($next = ($opener + 1); $next < $phpcsFile->numTokens; $next++) {
+                $code = $tokens[$next]['code'];
+
+                if ($code === T_WHITESPACE) {
+                    continue;
+                }
+
+                // Skip all empty tokens on the same line as the opener.
+                if ($tokens[$next]['line'] === $tokens[$opener]['line']
+                    && (isset(PHP_CodeSniffer_Tokens::$emptyTokens[$code]) === true
```

Comment:
Replace `PHP_CodeSniffer_Tokens` with `Tokens`
```suggestion
                    && (isset(Tokens::$emptyTokens[$code]) === true
```